### PR TITLE
Add the ability to pass config for multiple Dynamo tables to a single app

### PR DESCRIPTION
**Pull request by alexwlchan**
## What is this PR trying to achieve?

On the reindexer app, we need to be able to pass the name of more than one table – the reindex table, and the table we're reindexing. This modifies our config model to allow passing config for multiple tables. It follows on from #296, where we separated table config from AWS config.

Part of #194.

## Who is this change for?

Developers who want a reindexer!

## Have the following been considered/are they needed?

- [x] Tests?
- [x] Docs?
- [x] Spoken to the right people?

### CODE REVIEW

File: reindexer/src/test/scala/uk/ac/wellcome/platform/idminter/StartupTest.scala [hunk 1]
```diff
@@ -1,17 +0,0 @@
-package uk.ac.wellcome.platform.reindexer
```
Comment (kenoir) [inline reindexer/src/test/scala/uk/ac/wellcome/platform/idminter/StartupTest.scala]: We still need these tests I think _just in the right place_.

File: common/src/main/scala/uk/ac/wellcome/finatra/modules/DynamoConfigModule.scala [hunk 2]
```diff
@@ -3,20 +3,53 @@ package uk.ac.wellcome.finatra.modules
...
+
+  def appNameFlagId = applicationNameTpl.format(tableName)
+  def arnFlagId = arnTpl.format(tableName)
+  def tableFlagId = tableTpl.format(tableName)
+}
+
+object MiroTableDynamoConfigModule extends DynamoConfigModule {
+  val tableName = "miroData"
+
+  val applicationName = flag[String](appNameFlagId, "", "Name of the Kinesis app")
```
Comment (kenoir) [inline common/src/main/scala/uk/ac/wellcome/finatra/modules/DynamoConfigModule.scala]: I tried various ways of pushing this functionality down into the trait but to no avail.

File: common/src/main/scala/uk/ac/wellcome/finatra/modules/DynamoConfigModule.scala [hunk 3]
```diff
@@ -3,20 +3,53 @@ package uk.ac.wellcome.finatra.modules
...
+  val arn  = flag[String](arnFlagId, "", "ARN of the DynamoDB stream")
```
Comment (alexwlchan) [inline common/src/main/scala/uk/ac/wellcome/finatra/modules/DynamoConfigModule.scala]: nit: double space

File: common/src/main/scala/uk/ac/wellcome/finatra/modules/DynamoConfigModule.scala [hunk 4]
```diff
@@ -3,20 +3,53 @@ package uk.ac.wellcome.finatra.modules
...
+  val applicationName: Flag[String]
+  val arn: Flag[String]
+  val table: Flag[String]
 
   @Singleton
   @Provides
   def providesDynamoConfig(): DynamoConfig =
     DynamoConfig(applicationName(), arn(), table())
+
+  val applicationNameTpl:String  = "aws.dynamo.%s.streams.appName"
```
Comment (alexwlchan) [inline common/src/main/scala/uk/ac/wellcome/finatra/modules/DynamoConfigModule.scala]: nit: space before the colon

File: transformer/src/main/java/uk/ac/wellcome/platform/transformer/annotations/MiroKinesisClientLibConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package uk.ac.wellcome.platform.transformer.annotations;
```
Comment (kenoir) [inline transformer/src/main/java/uk/ac/wellcome/platform/transformer/annotations/MiroKinesisClientLibConfiguration.java]: Remove these.

File: calm_adapter/src/main/scala/uk/ac/wellcome/calm_adapter/modules/ActorRegistryModule.scala [hunk 6]
```diff
@@ -17,8 +17,8 @@ import uk.ac.wellcome.finatra.modules.{AkkaModule, DynamoClientModule, DynamoCon
 object ActorRegistryModule extends TwitterModule {
 
   override val modules = Seq(
-    DynamoClientModule,
     DynamoConfigModule,
+    DynamoClientModule,
```
Comment (alexwlchan) [inline calm_adapter/src/main/scala/uk/ac/wellcome/calm_adapter/modules/ActorRegistryModule.scala]: nit: superfluous change

File: common/src/main/scala/uk/ac/wellcome/finatra/modules/DynamoConfigModule.scala [hunk 7]
```diff
@@ -3,20 +3,56 @@ package uk.ac.wellcome.finatra.modules
...
+  val miroTableName = "miroData"
+
+  val miroApplicationName: Flag[String] = flag[String](appNameFlagId(miroTableName), "", "Name of the Kinesis app")
+  val miroArn: Flag[String] = flag[String](arnFlagId(miroTableName), "", "ARN of the DynamoDB stream")
+  val miroTable: Flag[String] = flag[String](tableFlagId(miroTableName), "", "Name of the DynamoDB table")
+
+  @Singleton
+  @Provides
+  @MiroDynamoConfig
+  def providesMiroDynamoConfig(): DynamoConfig =
```
Comment (kenoir) [inline common/src/main/scala/uk/ac/wellcome/finatra/modules/DynamoConfigModule.scala]: @alicefuzier you might be able to think of a better way to do this.

**Pull request merged**